### ISSUE TITLE
When I try to create and use a subworkflow, I need to restart the server for it to register

### ISSUE THREAD

**Issue by skamensky** [cortezaproject/corteza#991]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The issue is that

https://github.com/cortezaproject/corteza/blob/d868391798eb1a3bd1ef61afea9c926bd78da470/server/automation/service/workflow_converter.go#L561

Subworkflow uses `handleToID` which uses `svc.wIndex[h]` which  is only loaded once. in 
https://github.com/cortezaproject/corteza/blob/d868391798eb1a3bd1ef61afea9c926bd78da470/server/automation/service/workflow.go#L502

Instead I suggest we should either update `svc.wIndex[h]` as workflows aree added, or modify `handleToID` to use https://github.com/cortezaproject/corteza/blob/d868391798eb1a3bd1ef61afea9c926bd78da470/server/automation/service/workflow.go#L39 which is updated as workflows are added

**Comment by skamensky** [comment]
This seems like a pretty basic issue. Is no one else running into this?

Please let me know if you guys are interested in me turning this into a PR

**Comment by mlohoar** [comment]
I'd vote for this. I just wasted a day discovering this. Perhaps at least add something to the documentation to require a restart?

**Comment by tjerman** [comment]
It should be resolved with the next 2023.9 patch release.

```
## Fixed

### What

newly added sub-workflows failing to execute with a not found error

### How

by properly updating internal indexes

### Refs

2023.9.x monorepo `405ab1364`

```
